src/transformers/models/mbart/tokenization_mbart50.py

```diff
@@ -0,0 +1,235 @@
+# coding=utf-8
+# Copyright 2020 The Facebook AI Research Team Authors and The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from contextlib import contextmanager
+from typing import List, Optional
+
+from ...tokenization_utils import BatchEncoding
+from ...utils import logging
+from ..xlm_roberta.tokenization_xlm_roberta import XLMRobertaTokenizer
+
+
+logger = logging.get_logger(__name__)
+
+_all_mbart_models = ["facebook/mbart-50-large-one-to-many"]
+SPM_URL = "https://huggingface.co/facebook/mbart-50-large-one-to-many/resolve/main/sentencepiece.bpe.model"
+
+FAIRSEQ_LANGUAGE_CODES = [
+    "ar_AR",
+    "cs_CZ",
+    "de_DE",
+    "en_XX",
+    "es_XX",
+    "et_EE",
+    "fi_FI",
+    "fr_XX",
+    "gu_IN",
+    "hi_IN",
+    "it_IT",
+    "ja_XX",
+    "kk_KZ",
+    "ko_KR",
+    "lt_LT",
+    "lv_LV",
+    "my_MM",
+    "ne_NP",
+    "nl_XX",
+    "ro_RO",
+    "ru_RU",
+    "si_LK",
+    "tr_TR",
+    "vi_VN",
+    "zh_CN",
+    "af_ZA",
+    "az_AZ",
+    "bn_IN",
+    "fa_IR",
+    "he_IL",
+    "hr_HR",
+    "id_ID",
+    "ka_GE",
+    "km_KH",
+    "mk_MK",
+    "ml_IN",
+    "mn_MN",
+    "mr_IN",
+    "pl_PL",
+    "ps_AF",
+    "pt_XX",
+    "sv_SE",
+    "sw_KE",
+    "ta_IN",
+    "te_IN",
+    "th_TH",
+    "tl_XX",
+    "uk_UA",
+    "ur_PK",
+    "xh_ZA",
+    "gl_ES",
+    "sl_SI",
+]
+
+
+class MBart50Tokenizer(XLMRobertaTokenizer):
```

Comment:
Not sure we want to continue our inheritance pattern for tokenizers - this tokenizer looks quite different to XLMRoberta to me so I think I'd be in favor of not using inheritance here (wdyt @LysandreJik @sgugger ?)

Reply:
I agree, and MBart in itself isn't linked to XLM-R imo, so the inheritance makes it a bit hard to understand.

Reply:
Agreed with both of you. It should also take `src_lang` and `tgt_lang` in the init so that the attributes are created and initialized. The user can then change these attributes to change the languages used by the tokenzier.

That behavior should be documented.